yagi/src/main/java/org/raml/yagi/framework/grammar/rule/ErrorNodeFactory.java

```diff
@@ -142,9 +142,13 @@ public static Node createInvalidMinProperties(int minProperties)
         return new ErrorNode("Expected min properties " + minProperties);
     }
 
-    public static Node createInvalidMinLength(int minLength)
+    public static Node createInvalidMinLength(int minLength, Node node)
     {
-        return new ErrorNode("Expected min length " + minLength);
+        if (node.getParent() == null)
+            return new ErrorNode("Expected min length " + minLength + " for value \"" + node.toString() + "\"");
+        Node field = node.getParent().getChildren().get(0);
```

Comment:
Another way to get the field node, a more readable one, is casting the parent node to KeyValueNode and ask for the key, like this:

```
KeyValueNode parent = (KeyValueNode) node.getParent();
Node field parent.getKey();
```